### PR TITLE
A few suggestions

### PR DESCRIPTION
**Pull request by jrudolph**
I'll add a few commits with a few potential improvements. Use as much as you want to ;)

First one is the separation of PersistentBehavior API and implementation.

### CODE REVIEW

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedStashManagement.scala [hunk 1]
```diff
@@ -21,7 +21,6 @@ private[akka] trait EventsourcedStashManagement {
   protected def log: LoggingAdapter
 
   protected def extension: Persistence
-  protected def context: ActorContext[Any]
```
Comment (ktoso) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedStashManagement.scala:24]: yeah good I guess

**Comment by jrudolph** [comment]
Added another commit which consolidate Callbacks/PluginIds and PersistentBehaviorImpl into a single EventsourcedSetup which is then passed around.

**Review by ktoso — COMMENTED** [review]
looks good as well

**Comment by ktoso** [comment]
Shall I merge, should I keep all them separate commits?
Non controversial improvements, so thanks, happy to take those all

**Pull request merged**